src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -161,19 +161,45 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!-- Add conditional compilation symbols for the target framework (for example NET461, NETSTANDARD2_0, NETCOREAPP1_0) -->
   <PropertyGroup Condition=" '$(DisableImplicitFrameworkDefines)' != 'true' and '$(TargetFrameworkIdentifier)' != '.NETPortable' and '$(TargetFrameworkIdentifier)' != ''">
+    <_FrameworkVersionNumberForImplicitDefine>$(TargetFrameworkVersion.TrimStart('vV'))</_FrameworkVersionNumberForImplicitDefine>
+    <_FrameworkVersionForImplicitDefine>$(_FrameworkVersionNumberForImplicitDefine)</_FrameworkVersionForImplicitDefine>
     <_FrameworkIdentifierForImplicitDefine>$(TargetFrameworkIdentifier.Replace('.', '').ToUpperInvariant())</_FrameworkIdentifierForImplicitDefine>
     <VersionlessImplicitFrameworkDefine>$(_FrameworkIdentifierForImplicitDefine)</VersionlessImplicitFrameworkDefine>
+    <_FrameworkIdentifierForImplicitDefine Condition=" '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and '$(_FrameworkVersionForImplicitDefine)' >= '5.0' ">NET</_FrameworkIdentifierForImplicitDefine>
+    <VersionlessImplicitFrameworkDefine Condition=" '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and '$(_FrameworkVersionForImplicitDefine)' >= '5.0' ">$(VersionlessImplicitFrameworkDefine);$(_FrameworkIdentifierForImplicitDefine)</VersionlessImplicitFrameworkDefine>
```

Comment:
For separating property values? So we need to case on Visual Basic each time we add to this list?

Reply:
Check to make sure I'm right about it needing to be commas.  If so, try to look at how we handle it for other implicitly defined compilation constants.

A good way to handle it might be to create an `_ImplicitDefineConstant` item for each constant we want to add, and then have a target that runs that adds those to the `DefineConstants` property with the right syntax.